### ISSUE TITLE
Bug: "0-59/59 10 * * *" skips 10:00

### ISSUE THREAD

**Issue by pjebs** [dragonmantank/cron-expression#89]
@dragonmantank 

cron schedule: `0-59/59 10 * * *`

```php
$times = [];
$now = new Carbon();

$cron = CronExpression::factory("0-59/59 10 * * *");
for ($k = 0 ; $k < 20; $k++){
	$times[] = $cron->getNextRunDate($now, $k);
}
```

There is a bug. `2020-08-20 10:00:00` is missed. The first time is `2020-08-20 10:59:00`

**Comment by pjebs** [comment]
I'm using v2

**Comment by pjebs** [comment]
The bug seems to be caused by `/59`. It seems to be a special case because when I change it, it works.

On perhaps a related note: `0-58/65 10 * * *` seems to work but is invalid cron string.

**Comment by dragonmantank** [comment]
Weird, I'll dig into this. If it was a simple range/step issue, `0-12/12 * * * *` should exhibit the same behaviour (skipping the `00` minute) but it works as expected. 

I'll make an issue for the step being larger than the overall range.

**Comment by dragonmantank** [comment]
@pjebs Just to confirm, you expect this to consider both `10:00:00` and `10:59:00` as valid run times, correct?

**Comment by pjebs** [comment]
yes: https://www.netiq.com/documentation/cloud-manager-2-5/ncm-reference/data/bexyssf.html

**Comment by pjebs** [comment]
@dragonmantank Thank you for the library and also looking at my issue. I really appreciate you taking the initiative to take over prior project.

**Comment by dragonmantank** [comment]
Sorry this took so long, but this is now fixed in `master` and will go out soon.

The issue ended up being a simple `>=` vs `>` check for when we determine if we need to wrap around. So it was incorrectly assuming the final slot was beyond the range. 

The downside is that the wraparound functionality will go away. It doesn't affect this bug as a step of `59` is just the end, but if you relied on the wraparound functionality it will be gone in v4 as it breaks convention with the C implementation. I'll update this with the release version here in a bit.

**Comment by dragonmantank** [comment]
This should be fixed with v3.2.2
